nixos/modules/services/continuous-integration/buildbot/master.nix

```diff
@@ -247,6 +247,18 @@ in {
         description = lib.mdDoc "Packages to add the to the PYTHONPATH of the buildbot process.";
         example = literalExpression "pythonPackages: with pythonPackages; [ requests ]";
       };
+
+      environmentFiles = mkOption {
+        type = types.listOf types.path;
+        default = [];
+        example = [ "/run/secrets/buildbot-master" ];
+        description = lib.mdDoc ''
+          File to load as environment file. Environment variables from this file
+          will be interpolated into the config file using envsubst with this
+          syntax: `$ENVIRONMENT` or `''${VARIABLE}`.
```

Comment:
Mhm. I got one problem with that `master.cfg` contains python code and `{FOO}` is also a valid and common python construct. Given that we have python, Is `serviceConfig.LoadCredential` not enough?

In buildbot-nix, I hook up buildbot's secret engine with systemd's `LoadCredential` like this:

```
secrets.SecretInAFile(dirname=os.environ["CREDENTIALS_DIRECTORY"])
config["secretsProviders"].append(systemd_secrets)
```

For thinks like worker passwords, I use this:

```
def read_secret_file(secret_name: str) -> str:
    directory = os.environ.get("CREDENTIALS_DIRECTORY")
    if directory is None:
        print("directory not set", file=sys.stderr)
        sys.exit(1)
    return Path(directory).joinpath(secret_name).read_text().rstrip()
```

The advantage is that environment variables are not accidentally leaked to child processes as easy through environment variables and we don't perform sneaky text replacements.

Reply:
ahh thank you! I thought of using buildbot's secrets but did not know you could use it in things other than steps and services. But your solution is much better. Also I should've looked in buildbot-nix.

Reply:
actually read_secret_file works fine, but
```
c['secretsProviders'] = [
    secrets.SecretInAFile(dirname=os.environ["CREDENTIALS_DIRECTORY"]),
]
```
is the problematic part that checks for filemode of the secrets. But, don't need to use it as I can use the read_secret method, so for now I have it commented out. Still curious how you got it working!

Reply:
I fixed the filemode stuff in buildbot and it's part of nixpkgs unstable.
It was too hard to backport, but you can get the patch from here:

https://github.com/Mic92/buildbot-nix/blob/main/nix/0001-allow-secrets-to-be-group-readable.patch

Reply:
Thank you!